<div style="
  background: linear-gradient(90deg, #dfeffc, #e6f2ff);
  color: #003366;
  padding: 20px 30px;
  border-radius: 12px;
  border-left: 6px solid #0066cc;
  border-right: 6px solid #0066cc;
  box-shadow: 0 4px 12px rgba(0, 102, 204, 0.15);
  text-align: center;
  font-weight: bold;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  font-size: 35px;
  letter-spacing: 1px;
  margin: 40px auto 30px auto;
  width: fit-content;
">
🔹04. 🧹 Data Cleaning:
</div>

<div style="background-color: #e6f2ff; 
            #border-left: 6px solid #005b96; 
            padding: 12px 20px; 
            margin: 20px 0; 
            border-radius: 10px; 
            box-shadow: 0 2px 5px rgba(0,0,0,0.1); 
            font-family: 'Segoe UI', sans-serif; 
            font-size: 25px; 
            font-weight: 600; 
            color: #003366;">
  4.1 📌 Import Libraries:
</div>


In [4]:
# ------------------ 1. 📦 Custom Utilities ------------------
import MyLibrary as lib  # My custom helper functions (e.g., readfile)

# ------------------ 2. 📊 Data Handling ------------------
import pandas as pd
import numpy as np
import os

<div style="background-color: #fff3e0;
            #border-left: 5px solid #fb8c00;
            margin: 20px 30px;
            padding: 16px 24px;
            border-radius: 8px;
            font-family: 'Segoe UI', sans-serif;
            font-size: 18px;
            font-weight: 500;
            color: #e65100;">
  ✅ <strong>Conclusion:</strong> All essential Python libraries were successfully imported.<br><br>
  ▪️ <strong>📦 Custom Utilities:</strong> <code>MyLibrary</code> – Local helper functions (e.g., <code>readfile()</code>)<br>
  ▪️ <strong>📊 Data Handling:</strong> <code>pandas</code>, <code>numpy</code>, <code>os</code> – For data structures, operations, and file access<br><br>
  These imports form the foundation for loading, manipulating, and accessing dataset files throughout the notebook.
</div>

<div style="background-color: #e6f2ff; 
            #border-left: 6px solid #005b96; 
            padding: 12px 20px; 
            margin: 20px 0; 
            border-radius: 10px; 
            box-shadow: 0 2px 5px rgba(0,0,0,0.1); 
            font-family: 'Segoe UI', sans-serif; 
            font-size: 25px; 
            font-weight: 600; 
            color: #003366;">
  4.2 📌 Reading Dataset:
</div>


In [7]:
# I Will import function of readfile() from My Local own Library to read any file like csv file, excell file or etc.
data = lib.readfile("../Data/bankcustomer.csv")


Dataset of '../Data/bankcustomer.csv' loaded successfully.

Dataset Shape is : 10002 rows × 14 columns

Dataset Columns are :
 ['rownumber', 'customerid', 'surname', 'creditscore', 'geography', 'gender', 'age', 'tenure', 'balance', 'numofproducts', 'hascrcard', 'isactivemember', 'estimatedsalary', 'exited'] 




<div style="background-color: #fff3e0;
            #border-left: 5px solid #fb8c00;
            margin: 20px 30px;
            padding: 16px 24px;
            border-radius: 8px;
            font-family: 'Segoe UI', sans-serif;
            font-size: 18px;
            font-weight: 500;
            color: #e65100;">
  ✅ <strong>Conclusion:</strong> Dataset loaded successfully.<br><br>
  ▪️ File path: <code>../Data/bankcustomer.csv</code><br>
  ▪️ Dataset shape: <code>10002 rows × 14 columns</code><br>
  ▪️ Columns:<br>
  <code>
  ['rownumber', 'customerid', 'surname', 'creditscore', 'geography', 'gender', 'age', 'tenure',<br>
   'balance', 'numofproducts', 'hascrcard', 'isactivemember', 'estimatedsalary', 'exited']
  </code>
</div>


<div style="background-color: #e6f2ff; 
            #border-left: 6px solid #005b96; 
            padding: 12px 20px; 
            margin: 20px 0; 
            border-radius: 10px; 
            box-shadow: 0 2px 5px rgba(0,0,0,0.1); 
            font-family: 'Segoe UI', sans-serif; 
            font-size: 25px; 
            font-weight: 600; 
            color: #003366;">
  4.3 📌 Data Cleaning:
</div>


In [10]:
data = lib.DataCleaning(data,
    drop_columns={"rownumber","customerid","surname"},
    duplicated_values="drop", missing_values="drop",
    column_types={"hascrcard":"int","isactivemember":"int"},
    output_format="csv")

Dataset Report After Cleaning  |  Timestamp: 2025-08-05 23:00:23

Original dataset shape: 10002 rows, 14 columns

Dropped unnecessary columns: ['rownumber', 'customerid', 'surname']

Removed 2 duplicate rows.

Dropped 4 rows due to missing values in any column.

Outliers: Not specified (kept as is).

Changed column 'hascrcard' to type int

Changed column 'isactivemember' to type int

Cleaned dataset shape: 9996 rows, 11 columns

🧮 Rows difference: 6 rows removed during cleaning.
🧮 Columns difference: 3 columns removed during cleaning.

✅ Cleaned dataset saved to: ../Data\Cleaned_dataset.csv


<div style="background-color: #fff3e0;
            #border-left: 5px solid #fb8c00;
            margin: 20px 30px;
            padding: 16px 24px;
            border-radius: 8px;
            font-family: 'Segoe UI', sans-serif;
            font-size: 18px;
            font-weight: 500;
            color: #e65100;">
  ✅ <strong>Conclusion:</strong> Dataset cleaning completed successfully.<br><br>

  ▪️ <strong>Original dataset shape:</strong> <code>10002 rows × 14 columns</code><br>
  ▪️ <strong>Cleaned dataset shape:</strong> <code>9996 rows × 11 columns</code><br><br>

  ▪️ Dropped unnecessary columns: <code>['rownumber', 'surname', 'customerid']</code><br>
  ▪️ Removed <code>2</code> duplicate rows<br>
  ▪️ Dropped <code>4</code> rows due to missing values<br>
  ▪️ Outliers: <i>Not specified (kept as is)</i><br>
  ▪️ Converted column <code>'hascrcard'</code> to <code>int</code><br>
  ▪️ Converted column <code>'isactivemember'</code> to <code>int</code><br><br>

  🧮 <strong>Rows difference:</strong> 6 rows removed during cleaning<br>
  🧮 <strong>Columns difference:</strong> 3 columns removed during cleaning<br><br>

  💾 Cleaned dataset saved to: <code>../Data/Cleaned_dataset.csv</code><br>
  🕒 Timestamp: <strong>2025-08-04 00:49:41</strong>
</div>